backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/service/CatalogueSiteMap.java

```diff
@@ -87,4 +109,35 @@ private WebSitemapUrl urlForResource(
             resourceBasePath + "/all/" + resourcePath.name() + "/" + resourceId)
         .build();
   }
+
+  private WebSitemapUrl urlForVariable(Row variable) throws MalformedURLException {
+    String variableId = encodePathSegment(variable.getString("name"));
+    String resource = encodePathSegment(variable.getString(RESOURCE));
+    String dataset = encodePathSegment(variable.getString("dataset"));
+
+    // human-readable key
+    String variableCombiKey = String.join("-", variableId, resource, dataset, resource);
+
+    // key used to pinpoint the resource
+    String queryPart =
+        String.format(
+            "?keys={\"name\":\"%s\",\"resource\":{\"id\":\"%s\"},\"dataset\":{\"name\":\"%s\",\"resource\":{\"id\":\"%s\"}}}",
+            variableId, resource, dataset, resource);
+
+    return new WebSitemapUrl.Options(
+            "%s/all/variables/%s%s"
```

Comment:
I didn't realize we use keys. But this looks ok to me.

I think in next major revision of catalogue we would like to have a single nice path for all resources we publish
(but I don't know how to do the context stuff then, as in, what catalogue was used to enter, maybe that should become a 'get' variable then instead).